modules/porous_flow/src/materials/PorousFlow1PhaseP.C

```diff
@@ -0,0 +1,152 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "PorousFlow1PhaseP.h"
+
+template<>
+InputParameters validParams<PorousFlow1PhaseP>()
+{
+  InputParameters params = validParams<PorousFlowVariableBase>();
+
+  params.addRequiredCoupledVar("porepressure", "Variable that represents the porepressure of the single phase");
+  params.addClassDescription("This Material is used for the fully saturated single-phase situation where porepressure is the primary variable");
+  return params;
+}
+
+PorousFlow1PhaseP::PorousFlow1PhaseP(const InputParameters & parameters) :
+    PorousFlowVariableBase(parameters),
+
+    _porepressure_nodal_var(coupledNodalValue("porepressure")),
+    _porepressure_qp_var(coupledValue("porepressure")),
+    _gradp_qp_var(coupledGradient("porepressure")),
+    _porepressure_varnum(coupled("porepressure"))
+{
+  if (_dictator_UO.numPhases() != 1)
+    mooseError("The Dictator proclaims that the number of phases is " << _dictator_UO.numPhases() << " whereas PorousFlow1PhaseP can only be used for 1-phase simulations.  Be aware that the Dictator has noted your mistake.");
+}
+
+void
+PorousFlow1PhaseP::initQpStatefulProperties()
+{
+  PorousFlowVariableBase::initQpStatefulProperties();
+
+  /*
+   *  YAQI HACK !!
+   *
+   * I really just want to simply call
+   * buildQpPPSS();
+   * but i do it below in computeQpProperties
+   */
+  buildQpPPSS();
+}
+
+void
+PorousFlow1PhaseP::computeQpProperties()
+{
+  buildQpPPSS();
+
+  /*
+   *  YAQI HACK !!
+   *
+   * I really just want to simply call
+   * buildQpPPSS();
+   * from initQpStatefulProperties, but the Variables
+   * aren't initialised at that point so moose crashes
+   */
+  /*if (_t_step == 1)
+    for (unsigned ph = 0; ph < _num_ph; ++ph)
+    {
+      _porepressure_old[_qp][ph] = _porepressure[_qp][ph];
+      _saturation_old[_qp][ph] = _saturation[_qp][ph];
+    }
+  */
+
+  // prepare the derivative matrix with zeroes
+  for (unsigned phase = 0; phase < _num_phases; ++phase)
+  {
+    _dporepressure_nodal_dvar[_qp][phase].assign(_dictator_UO.numVariables(), 0.0);
```

Comment:
don;t you want to create a `const unsigned int` with `_dictator_UO.numVariables()` so you don't repeatedly have the call in your code?